src/firstboot/firstboot.c

```diff
@@ -309,28 +323,34 @@ static int prompt_locale(void) {
         return 0;
 }
 
-static int process_locale(void) {
-        const char *etc_localeconf;
+static int process_locale(int rfd) {
+        _cleanup_close_ int pfd = -EBADF;
+        _cleanup_free_ char *f = NULL;
         char* locales[3];
         unsigned i = 0;
         int r;
 
-        etc_localeconf = prefix_roota(arg_root, "/etc/locale.conf");
-        if (laccess(etc_localeconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming locale information has been configured.",
-                          etc_localeconf);
-                return 0;
-        }
+        assert(rfd >= 0);
+
+        pfd = chase_and_open_parent_at(rfd, "/etc/locale.conf",
+                                       CHASE_AT_RESOLVE_IN_ROOT|CHASE_MKDIR_0755|CHASE_WARN,
+                                       &f);
+        if (pfd < 0)
+                return log_error_errno(pfd, "Failed to chase /etc/locale.conf: %m");
 
-        if (arg_copy_locale && arg_root) {
+        r = should_configure(pfd, f);
+        if (r == 0)
+                log_debug("Found /etc/locale.conf, assuming locale information has been configured.");
+        if (r <= 0)
+                return r;
 
-                (void) mkdir_parents(etc_localeconf, 0755);
-                r = copy_file("/etc/locale.conf", etc_localeconf, 0, 0644, COPY_REFLINK);
+        if (arg_copy_locale && !dir_fd_is_root(rfd)) {
```

Comment:
hmm, dir_fd_is_root() might fail and the error is eaten up. ideally we'd log about unexpected errors on that. But at least I'd prefer if we do not suggest the return value of the func was boolean here, and thus do a numeric check rather than than a boolean check to handle errors.

i.e. `dir_fd_is_root(root_fd) == 0` or so. matching our coding style this makes clear the return value is not boolean but numeric, but we treat errors like true.